monai/apps/utils.py

```diff
@@ -327,7 +327,17 @@ def download_and_extract(
             be False.
         progress: whether to display progress bar.
     """
+    urlFilenameExtension = ''.join(Path(".", _basename(url)).resolve().suffixes)
+    if filepath:
+        FilepathExtenstion = ''.join(Path(".", _basename(filepath)).resolve().suffixes)
+        if urlFilenameExtension != FilepathExtenstion:
+            raise NotImplementedError(
```

Comment:
For reference, `NotImplementedError` indicates some method or something isn't implemented, it's not meant for this kind of error, so instead you should use `ValueError` or some other things more closely aligned with this sort of condition.